core/src/main/java/greencity/controller/GoalController.java

```diff
@@ -57,50 +58,61 @@ public GoalController(GoalService goalService) {
     }
 
     /**
-     * Method returns shopping list by user id.
+     * The controller which saveGoal {@link Goal}.
      *
-     * @return shopping list {@link ShoppingListDtoResponse}.
-     * @author Marian Datsko
+     * @param goalPostDto {@link GoalDto}
+     * @return {@link ResponseEntity}
      */
-    @ApiOperation(value = "Get shopping list")
+    @ApiOperation(value = "Save goal")
     @ApiResponses(value = {
         @ApiResponse(code = 200, message = HttpStatuses.OK),
         @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
-        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
     })
-    @GetMapping("/shoppingList/{userId}")
-    @ApiLocale
-    public ResponseEntity<List<ShoppingListDtoResponse>> getShoppingList(
-        @ApiParam("User id")
-        @PathVariable Long userId,
-        @ApiIgnore @ValidLanguage Locale locale) {
-        return ResponseEntity.status(HttpStatus.OK).body(goalService.getShoppingList(userId, locale.getLanguage()));
+    @PostMapping
+    public ResponseEntity<List<LanguageTranslationDTO>> save(@Valid @RequestBody GoalPostDto goalPostDto) {
+        List<LanguageTranslationDTO> response = mapper.map(goalService.saveGoal(goalPostDto),
+            new TypeToken<List<LanguageTranslationDTO>>() {
+            }.getType());
+        return ResponseEntity.status(HttpStatus.CREATED).body(response);
+    }
+
+    /**
+     * The controller which update {@link GoalTranslation}.
+     *
+     * @param goalPostDto {@link GoalPostDto}
+     * @return {@link ResponseEntity}
+     */
+    @ApiOperation(value = "Update goal")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @PutMapping("/{id}")
+    public ResponseEntity<List<LanguageTranslationDTO>> update(
+        @Valid @RequestBody GoalPostDto goalPostDto) {
+        List<LanguageTranslationDTO> response = mapper.map(goalService.update(goalPostDto),
```

Comment:
Move mapping operation to service method. Controller is responsible for receiving request and sending response - mapping should be done in service